Implement findAssets api

natebosch opened the pull request:
- Implement findAssets based on package path in BazelFileSystem
- Add a rootPackageName field to BazelAssetReader - it is necessary to
  reconstruct an AssetId for paths within the root package
- Implement findAssets in BazelAssetReader
- Add a simple test for findAssets
- Bump to build 0.8.0

natebosch committed:
Implement findAssets api

- Implement findAssets based on package path in BazelFileSystem
- Add a rootPackageName field to BazelAssetReader - it is necessary to
  reconstruct an AssetId for paths within the root package
- Implement findAssets in BazelAssetReader
- Add a simple test for findAssets
- Bump to build 0.8.0

natebosch commented:
Will need to publish package:build for travis to be happy

jakemac53 commented on bazel_codegen/test/asset_reader_test.dart:
```diff
@@ -49,6 +50,12 @@ void main() {
   });
+
+  test('findAssets', () async {
+    fileSystem.nextFindAssets = ['lib/foo.dart'];
```
Lets add more than one here even though it doesn't make much of a difference

natebosch replied:
Done

jakemac53 commented on bazel_codegen/lib/src/assets/file_system.dart:
```diff
@@ -32,6 +33,17 @@ class BazelFileSystem {
   String readAsStringSync(String path, {Encoding encoding: UTF8}) =>
       _fileForPath(path).readAsStringSync(encoding: encoding ?? UTF8);
 
+  Iterable<String> findAssets(String packagePath, Glob glob) sync* {
```
Lets add a test for this as well, should be able to just create a real tmp dir or something with some files and provide that as a search path.

natebosch replied:
Done

natebosch committed:
Add test for file_system findAssets

- Add dependency on test_descriptor to create test directory
- Add test for findAssets in BazelFileSystem
- Expand test for findAssets in BazelAssetReader to include multiple
  files

jakemac53 commented on bazel_codegen/pubspec.yaml:
```diff
@@ -19,3 +19,8 @@ dependencies:
+
+dependency_overrides:
+  build:
+    path: ../../pkg-build/build
```
delete?

natebosch replied:
oops!

natebosch committed:
Delete dependency overrides
